src/autoscaler/metricscollector/collector/apppoller.go

```diff
@@ -61,10 +61,11 @@ func (ap *appPoller) Stop() {
 func (ap *appPoller) startPollMetrics() {
 	for {
 		ap.pollMetric()
+		ap.timer.Reset(ap.pollInterval)
```

Comment:
This is broken like I have discussed. Just stop the previous timer and create a new one.
Don't stop it in Stop().